content/en/docs/architecture/component-overview.md

```diff
@@ -183,6 +183,175 @@ return
 
+#### Device Metadata
+
+{{< plantuml id="device-metadata" >}}
+@startuml Sequence
+skinparam backgroundColor transparent
+hide footbox
+
+participant D as "Device"
+participant CGW as "CoAP Gateway"
+participant RA as "Resource Aggregate"
+participant RD as "Resource Directory"
+participant EB as "Event Bus"
+
+group Device has signed in
+   D -> CGW ++: Sign In
+   activate D
+   CGW -> RA++: Update device to online and twin synchronization to out of sync or disabled state
+   RA -> EB: Publish device metadata updated event
+   RA->CGW:
+   deactivate RA
+   CGW -> D: Signed In
+   deactivate D
+   CGW -> D++: Get discovery resource (oic/res)
+   return resources links
+   alt batch observation is supported and twin is enabled
+      CGW -> RA++: Update twin synchronization to syncing state
+      RA -> EB: Publish device metadata updated event
+      RA -> CGW
+      deactivate RA
+      CGW -> D++: Observe discovery resource (oic/res) via batch observation
+      D -> CGW: first notification from observe
+      CGW -> RA++: notify resources are changed
+      return
+      CGW -> RA++: Update twin synchronization to in sync state
+      RA -> EB: Publish device metadata updated event
+      RA -> CGW
+      deactivate RA
+      deactivate D
+   else batch observation is not supported and twin is enabled
+      CGW -> RD++: Get published resources
+      return published resources
+      CGW -> RA++: Update twin synchronization to syncing state
+      RA -> EB: Publish device metadata updated event
+      RA->CGW:
+      deactivate RA
+      loop for each new published resource
+         CGW -> D++: Observe resource
+         D -> CGW: first notification from observe
+         CGW -> RA++: notify resources are changed
+         return
+      end
+      CGW -> RA++: Update twin synchronization to in sync state
+      RA -> EB: Publish device metadata updated event
+      RA->CGW:
+      deactivate RA
+      deactivate D
+   end
+   deactivate CGW
+end
+
+group Device publish resources
+  D -> CGW ++: Publish resources
+  activate D
+  CGW -> D:
+  deactivate D
+  alt batch observation is supported and batch observation is not created and twin is enabled
```

Comment:
What is this?

Reply:
When moving diagrams to device-twin.md, please make sure you also separate completely to different diagrams cases when batch is enabled and when not, because either user have it or doesn't have it enabled. In case he has it, what's the default, the diagram is full of information not relevant for him. So one paragraph is ### when it's enabled and one when it's disabled. Don't make alts for this in plantuml.